specs/drafts/genesis-v2.md

```diff
@@ -0,0 +1,174 @@
+      ## - 'federated': members and weights are determined arbitrarily by an authority at the parent (proof of authority style).
+      mode: static | collateral | federated
+      ## The scaling factor that applies to the raw power communicated to the parent (raw * 1e-{power_scale}). 
+      ## To be deprecated during contracts refactor project.
+      power_scale: uint8
+      ## The maximum number of active validators at a time.
+      ## To be deprecated during contracts refactor project.
+      max_active: uint64
+    ## Initial validator set as an array of pubkeys and weights.
+    initial_members:
+      - pubkey: bytes
+        weight: uint64
+  
+## Balances to be credited to the specified addresses at genesis, directly on the state tree,
+## the precision unit of the circulating supply source.
+balances:
+  [address]: uint256
+ 
+## Execution layer configuration. 
+execution:
+  ## Gas market configuration
+  gas_market:
+    ## Oscillation function. Values: eip1559 or static.
+    kind: eip1559 | static
+    ## Initial base fee.
+    initial_base_fee: uint256 
+  ## Configuration of FVM runtimes.
+  runtimes:
+    ## Filecoin EVM configuration.
+    fevm:
+      ## Contract deployment permissions.
+      contract_deployment_permissions:
+        ## Contract deployment permissioning mode. Values: 'unrestricted', or 'allowlist'.
+        mode: unrestricted | allowlist
+        ## If the mode is 'allowlist', specify a list of allowed addresses, or blank if none is allowed.
+        allowlist:
+          - addresses
+         
+## CIDs of actor manifests for built-in actors and for custom actors.
+actor_bytecode:
+  builtin: Cid
+  custom: Cid | null
+
+## The IPC system contracts, as a CID to the root of a UnixFS DAG stored in the inline blockstore.
+system_contracts: Cid
+
+## The initial state tree after
+initial_state_tree: Cid
+
+## Blockstore serialized as a compressed inline CAR.
+blockstore: inline_bytes
+```
+
+### Initializing a genesis specification
+
+A genesis specification can be initialized in one of two ways, depending on the kind of subnet.
+
+For a child subnet, we use the `ipc genesis spec init-child` tool. This tool takes the parent subnet RPC, the gateway
+address, and the subnet ID. It initializes a prefilled genesis, populating as much detail as possible from the on-chain
+subnet definition.
+
+For a standalone (unrooted) subnet, we use the `ipc genesis spec init-standalone --template=<name>` command. This tool
+simply writes an empty genesis based on the specified template, for the user to adjust manually.
+
+### Customizing a genesis specification
+
+For the most part, developers customize the genesis specification by manually editing the file, except for the
+following elements:
+
+- actor_bytecode/builtin
+- actor_bytecode/custom
+- system_contracts
+- initial_state_tree
+- blockstore
+
+Actor bytecode and system contracts are added using the `ipc genesis spec update` tool one or several of the following
```

Comment:
The update is not incremental update right? Write all for each call.

Reply:
I didn't follow your question entirely, but the idea is that whatever thing you specify will overwrite the previous value for that thing. So if you set builtin actors bytecode, and then you go and call this command again, it will overwrite the previous value for that field. And it will maintain the blockstore succinct.